rcl_lifecycle/src/rcl_lifecycle.c

```diff
@@ -285,15 +285,22 @@ rcl_lifecycle_is_valid_transition(
   const rcl_lifecycle_state_t * current_state = rcl_lifecycle_get_state(
     &state_machine->transition_map, current_id);
 
+  if (NULL == current_state) {
```

Comment:
nitpick: I'd suggest either use `if (!current_state)` or also apply the `RCL_CHECK_FOR_NULL_WITH_MSG` macro here.